module/userExtension/helper.js

```diff
@@ -211,304 +212,373 @@ module.exports = class UserExtensionHelper {
+              user = await database.models.userExtension.create(userInformation);
+              userExtensionMap[user.userId.toString()] = user;
+
+              userRole['_SYSTEM_ID'] = user?._id || '';
+              userRole.status = user ? 'Success' : 'Failed to create the user role.';
+              userRole._kafkaEventPayloads = kafkaEventPayloads;
+            } else {
+              // Update existing user
+              let existingUserPlatformRoles = [...(existingUser.platformRoles || [])];
 
-                if (existingUser.platformRoles && existingUser.platformRoles.length > 0) {
-                  userPlatformRoleToUpdate = _.findIndex(existingUser.platformRoles, { code: userRole.platform_role });
-                }
+              if (userRole.programOperation && userRole.programs) {
+                for (const program of userRole.programs) {
+                  const programObjectId = programIdMap[program];
 
-                if (!(userPlatformRoleToUpdate >= 0)) {
-                  userPlatformRoleToUpdate = existingUser.platformRoles.length;
-                  userRoleMap[userRole.platform_role].programs = new Array();
-                  existingUser.platformRoles.push(userRoleMap[userRole.platform_role]);
-                }
+                  if (!programObjectId) {
+                    userRole.status = messageConstants.apiResponses.PROGRAM_NOT_FOUND;
+                    userRolesUploadedData.push(userRole);
+                    continue outerloop;
+                  }
 
-                existingUser.platformRoles[userPlatformRoleToUpdate].programs = existingUser.platformRoles[
-                  userPlatformRoleToUpdate
-                ].programs.map((program) => program.toString());
-
-                if (userRole.programOperation == 'OVERRIDE') {
-                  existingUser.platformRoles[userPlatformRoleToUpdate].programs = programIds;
-                } else if (userRole.programOperation == 'APPEND' || userRole.programOperation == 'ADD') {
-                  existingUser.platformRoles[userPlatformRoleToUpdate].programs =
-                    existingUser.platformRoles[userPlatformRoleToUpdate].programs.concat(programIds);
-                  existingUser.platformRoles[userPlatformRoleToUpdate].programs = _.uniq(
-                    existingUser.platformRoles[userPlatformRoleToUpdate].programs,
+                  const currentRoleInfoIndex = existingUserPlatformRoles.findIndex(
+                    (pr) => pr.programId.toString() === programObjectId.toString()
                   );
-                } else if (userRole.programOperation == 'REMOVE') {
-                  if (programIds.length > 0) {
-                    programIds.forEach((programId) => {
-                      _.pull(existingUser.platformRoles[userPlatformRoleToUpdate].programs, programId.toString());
-                    });
+
+                  const newRoles = platform_role_array;
+
+                  if (userRole.programOperation === 'OVERRIDE') {
```

Comment:
@borkarsaish65 define common constants for OVERRIDE, ADD, APPEND, REMOVE etc . use them instead directly using string